src/server/project.ts

```diff
@@ -989,9 +1004,17 @@ namespace ts.server {
                 this.lastCachedUnresolvedImportsList = undefined;
             }
 
+            const isFirstLoad = this.projectProgramVersion === 0;
             if (hasNewProgram) {
                 this.projectProgramVersion++;
             }
+            if (hasAddedorRemovedFiles && this.autoImportProviderHost) {
+                this.autoImportProviderHost.markAsDirty();
+            }

```

Comment:
```suggestion
            if (hasAddedorRemovedFiles) {
                if (!this.autoImportProviderHost) this.autoImportProviderHost = undefined;
                this.autoImportProviderHost?.markAsDirty();
            }
```

Reply:
You would also want test the condition where your project contains everything from package json so no auto import provider and then you update your project code to not import one of the dependency.